lsif-java/src/main/scala/com/sourcegraph/lsif_java/commands/SnapshotLsifCommand.scala

```diff
@@ -59,7 +59,10 @@ case class SnapshotLsifCommand(
       in = AbsolutePath.of(inputPath, sourceroot)
       doc <- SnapshotLsifCommand.parseTextDocument(in, sourceroot)
     } {
-      SnapshotCommand.writeSnapshot(doc, finalOutput)
+      val docPath = AbsolutePath.of(Path.of(doc.getUri), sourceroot).toRealPath()
+      if (docPath.toAbsolutePath.startsWith(sourceroot)) {
```

Comment:
FYI, you can define variables and add if conditions inside for comprehensions. Not relevant in this diff here but could be useful if you want to continue to drill deeper into a data structure